core/src/main/java/jenkins/util/Timer.java

```diff
@@ -29,7 +29,19 @@
      * The scheduled executor thread pool. This is initialized lazily since it may be created/shutdown many times
      * when running the test suite.
      */
-    private static ScheduledExecutorService executorService;
+    static ScheduledExecutorService executorService;
+
+
+    /** Avoids issues where the Timer threads are initialized in a context where they can receive a bogus contextClassLoader.
+     *  @see <a href="https://issues.jenkins-ci.org/browse/JENKINS-49206">JENKINS-49206</>
+     */
+    static class ClassloaderSanityDaemonThreadFactory extends DaemonThreadFactory {
```

Comment:
Might want to make this an API in the same package as the others, using the delegate pattern like the others. Would be helpful e.g. for `workflow-support`’s `Timeout`.

Reply:
cf. https://github.com/jenkinsci/workflow-support-plugin/pull/53

Reply:
That's a fair point

Reply:
Done, @jglick